pkg/xds/server/components.go

```diff
@@ -66,9 +68,16 @@ func DefaultDataplaneSyncTracker(rt core_runtime.Runtime, reconciler SnapshotRec
 					},
 				}
 
+				outbound, err := xds_topology.GetOutboundTargets(ctx, dataplane, rt.ResourceManager())
```

Comment:
I see we do it for every dataplane every X seconds. This would be a bottleneck very soon (even with caching on resource store).

What I imagine is that we've got a job (Component=Runnable) that builds `Cluster` and `ClusterLoadAssignment` from `Dataplanes` and expose it to the `OutboundProxyGenerator`.
It's more efficient in CPU, because we do the matching once (every X second) and then `OutboundProxyGenerator` can just retrieve this in O(1).
It's more efficient in memory. Every Snapshot will be stored in SnapshotCache. I don't know about Go but when I introduced reusing protos in java-control-plane based control plane, there performance went through the roof
https://github.com/envoyproxy/java-control-plane/pull/98